### ISSUE TITLE
Command line copy not working for bash with powerline or zsh

### ISSUE THREAD

**Issue by infokiller** [tmux-plugins/tmux-yank#48]
The command line copy copies my full powerline prompt with all the unicode glyphs, and doesn't copy the command correctly. Do I need to configure something so that it works?

**Comment by aleprovencio** [comment]
I don't have powerline and it's not working also. Tryed with both xsel or xclip installed and the output after `prefix+y` is `^A^E` appended after the line content and filling my primary register with an `A`.

**Comment by aleprovencio** [comment]
OK I got it. As I use bash vi-keybindings this only works when not on insert-mode otherwise the above will happen. In addiction it copies from where cursor is until the end of the line, should't it copy the whole line just like vim's `yy`?

@infokiller, please let me know if this should be a new issue instead of noise to your original question.

**Comment by bruno-** [comment]
Hi,
@infokiller are you maybe using tmux via ssh?

@aleprovencio , it should copy the \*whole* line correctly. Can you make sure pressing `ctrl-a` in the command line moves the cursor to the beginning of the line?

**Comment by aleprovencio** [comment]
it does not @bruno, `^A` is printed, but as I mentioned this is `bash` on `editing-mode vi`.

**Comment by bruno-** [comment]
> this is `bash` on `editing-mode vi`

I missed that one. Bash `vi` mode is not supported.

**Comment by infokiller** [comment]
Sorry for the late reply, I missed this somehow!
No, I'm not using tmux via ssh. 
I'm using more zsh than bash these days and it also doesn't work for me there, without even a custom prompt. Instead of copying just the current command, it copies commands over multiple lines including the prompt characters.

**Comment by infokiller** [comment]
@bruno , @aleprovencio : any idea?

Thanks!

**Comment by IngoHeimbach** [comment]
As a workaround for zsh users, you can pass `<prefix>`+y to zsh and bind it to a function that copies the current zle content to clipboard. To get that running, insert a `bind-key` statement at the bottom of your `.tmux.conf` (note that ` is my prefix key):
```
# Initialize TMUX plugin manager
run '~/.tmux/plugins/tpm/tpm'

# send `y to zsh to copy current zle line to clipboard
bind-key y send-keys `y
```
Then, add the following your `.zshrc` (for macOS users):
```
function _copy-to-clipboard {
    print -rn -- $BUFFER | pbcopy
    [ -n "$TMUX" ] && tmux display-message 'Line copied to clipboard!'
}
zle -N _copy-to-clipboard
bindkey -M viins "\`y" _copy-to-clipboard
```
Linux users can simply replace `pbcopy` with `xclip`.

**Comment by docwhat** [comment]
For the basic issue of <kbd>prefix</kbd>–<kbd>y</kbd> in normal mode:

This should be pretty much fixed in `tmux` 2.4+ with to commit e01284962106089d39cbbff116a11dbfae19713f

@IngoHeimbach`s idea sounds pretty cool though complicated and tricky.

I'd be willing to look at a PR if desired.
